src/v/kafka/server/handlers/produce.cc

```diff
@@ -509,8 +509,12 @@ produce_topic(produce_ctx& octx, produce_request::topic& topic) {
             continue;
         }
 
-        const auto& kafka_noproduce_topics
+        auto kafka_noproduce_topics
           = config::shard_local_cfg().kafka_noproduce_topics();
+        if (!octx.rctx.authorized_auditor()) {
+            kafka_noproduce_topics.push_back(
+              model::kafka_audit_logging_topic());
+        }
```

Comment:
can we not just do `is_noproduce_topic |= !authorized_auditor()`

Reply:
I feel like we need something stronger than just depending on the `kafka_noproduce_topics` list.  This list is modifiable by any superuser of the cluster, meaning even if we stick in the audit log into this configuration, someone could come around and remove it.

An alternative idea is to prevent anyone from remove it from the list by modifying the handler for this configuration item.

Reply:
I don't think Noah is saying to depend on that, we can just extend the conditional of when to return the `not_authorized` response